metadata-io/src/main/java/com/linkedin/metadata/entity/EntityServiceImpl.java

```diff
@@ -996,49 +1002,56 @@ This condition is specifically for an older conditional write ingestAspectIfNotP
 
                             return result;
                           })
+                      .filter(Objects::nonNull)
                       .collect(Collectors.toList());
 
-              // commit upserts prior to retention or kafka send, if supported by impl
-              if (txContext != null) {
-                txContext.commitAndContinue();
-              }
-              long took = TimeUnit.NANOSECONDS.toMillis(ingestToLocalDBTimer.stop());
-              if (took > DB_TIMER_LOG_THRESHOLD_MS) {
-                log.info("Ingestion of aspects batch to database took {} ms", took);
-              }
-
-              // Retention optimization and tx
-              if (retentionService != null) {
-                List<RetentionService.RetentionContext> retentionBatch =
-                    upsertResults.stream()
-                        // Only consider retention when there was a previous version
-                        .filter(
-                            result ->
-                                batchAspects.containsKey(result.getUrn().toString())
-                                    && batchAspects
-                                        .get(result.getUrn().toString())
-                                        .containsKey(result.getRequest().getAspectName()))
-                        .filter(
-                            result -> {
-                              RecordTemplate oldAspect = result.getOldValue();
-                              RecordTemplate newAspect = result.getNewValue();
-                              // Apply retention policies if there was an update to existing aspect
-                              // value
-                              return oldAspect != newAspect
-                                  && oldAspect != null
-                                  && retentionService != null;
-                            })
-                        .map(
-                            result ->
-                                RetentionService.RetentionContext.builder()
-                                    .urn(result.getUrn())
-                                    .aspectName(result.getRequest().getAspectName())
-                                    .maxVersion(Optional.of(result.getMaxVersion()))
-                                    .build())
-                        .collect(Collectors.toList());
-                retentionService.applyRetentionWithPolicyDefaults(opContext, retentionBatch);
+              if (!upsertResults.isEmpty()) {
+                // commit upserts prior to retention or kafka send, if supported by impl
+                if (txContext != null) {
+                  txContext.commitAndContinue();
+                }
+                log.info(
+                    "Ingestion of aspects batch to database took {} in {} ms",
```

Comment:
Is it intentional to revert #11714 here?

Reply:
No, this was a bad merge from another branch. Good catch!